### PR TITLE
Add support to the `make` module for targets with options

### PR DESCRIPTION
**Pull request by stevekuznetsov**
##### ISSUE TYPE
- Feature Pull Request
##### COMPONENT NAME

`make`
##### ANSIBLE VERSION

```
$ ansible --version
ansible 2.2.0.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```
##### SUMMARY

In order to run a `make` target with options, we need to not
assume that the target that is passed in to the module is to
be interpreted as one command-line argument. This allows us
to support targets with options like:
  $ make some-target -o some-file

Where the `some-target` target skips rebuilding `some-file`
even if the file is old. Currently, the implementation would
attempt to run the `some-target -o some-file` target, which
does not exist.

`make` does not allow for targets with spaces in their names,
so we should be safe to break up the command-line arguments
like this and not risk mutilating a valid target name.

Signed-off-by: Steve Kuznetsov skuznets@redhat.com
###### BEFORE

```
TASK [make : execute make target(s) "test -o build" in the origin-web-console repository] ***
failed: [openshiftdevel] (item=test -o build) => {
    "cmd": "/bin/make 'test -o build'", 
    "failed": true, 
    "invocation": {
        "module_args": {
            "chdir": "/data/src/github.com/openshift/origin-web-console", 
            "params": null, 
            "target": "test -o build"
        }, 
        "module_name": "make"
    }, 
    "item": "test -o build", 
    "msg": "make: *** No rule to make target 'test -o build'.  Stop.", 
    "rc": 2, 
    "stderr": "make: *** No rule to make target 'test -o build'.  Stop.\n", 
    "stdout": "", 
    "stdout_lines": []
}
```
###### AFTER

```
TASK [make : execute make target(s) "test -o build" in the origin-web-console repository] ***
failed: [openshiftdevel] (item=test -o build) => {
    "cmd": "/bin/make test -o build", 
    "failed": true, 
    "invocation": {
        "module_args": {
            "chdir": "/data/src/github.com/openshift/origin-web-console", 
            "params": null, 
            "target": "test -o build"
        }, 
        "module_name": "make"
    }, 
    "item": "test -o build", 
    "msg": "<snip>make: *** [test] Error 3", 
    "rc": 2, 
    "stderr": "<snip>make: *** [test] Error 3\n", 
    "stdout": "<snip>Makefile:45: recipe for target 'test' failed\n", 
    "stdout_lines": [
        <snip>
        "Makefile:45: recipe for target 'test' failed"
    ]
}
```

(example after has a failed `make` target, but it runs correctly, just that example failed in the middle of the run)

@LinusU @gregdek  I've added the version comment in the documentation.

### CODE REVIEW

File: lib/ansible/modules/system/make.py [hunk 1]
```diff
@@ -34,7 +34,7 @@
 options:
   target:
     description:
-      - The target to run
+      - The target to run. As of Ansible 3.2, options can be specified as well.
```
Comment (resmo) [inline lib/ansible/modules/system/make.py]: Ansible 3.2? we are not there yet 😂

Comment (stevekuznetsov) [reply lib/ansible/modules/system/make.py]: I'll blame that one on dyslexia :)

**Comment by stevekuznetsov** [comment]
@jimi-c do you know why this `needs_revision`?

**Comment by stevekuznetsov** [comment]
Closing for inactivity ...

**Pull request closed without merging**